cache/memcached_client.go

```diff
@@ -196,13 +203,16 @@ func newMemcachedClient(
 		addressProvider: addressProvider,
 		stop:            make(chan struct{}, 1),
 		getMultiGate: gate.New(
-			prometheus.WrapRegistererWithPrefix("memcached_getmulti_", reg),
+			promregistry.TeeRegisterer{
```

Comment:
I don't think this is correct. The final metric name will start with `getmulti_` while it should be `(cache|memcached)_getmulti_*`.

Reply:
In my test, it gives me the correct result. Examples:
```
versioned_client_cache_getmulti_gate_queries_concurrent_max{backend="redis",name="admin-cache"} 100
thanos_cache_getmulti_gate_queries_in_flight{backend="memcached",component="query-frontend",name="frontend-cache"} 0
```

So from what I can see, `prometheus.WrapRegisterWithPrefix` will prepend passed registery and add prefix for subsequent wrap or for metrics which will use the Registerer.

I am building a simple program to verify:
```
package main

import (
	"net/http"
	"time"

	"github.com/prometheus/client_golang/prometheus"
	"github.com/prometheus/client_golang/prometheus/promauto"
	"github.com/prometheus/client_golang/prometheus/promhttp"
)

var (
	opsProcessed prometheus.Counter
)

func main() {
	reg := prometheus.NewRegistry()
	reg1 := prometheus.WrapRegistererWithPrefix("one_", reg)
	reg2 := prometheus.WrapRegistererWithPrefix("two_", reg1)
	reg3 := prometheus.WrapRegistererWithPrefix("three_", reg2)

	opsProcessed = promauto.With(reg3).NewCounter(prometheus.CounterOpts{
		Name: "myapp_processed_ops_total",
		Help: "The total number of processed events",
	})
	go func() {
		for {
			opsProcessed.Inc()
			time.Sleep(2 * time.Second)
		}
	}()
	http.Handle("/metrics2", promhttp.HandlerFor(reg, promhttp.HandlerOpts{Registry: reg3}))
	http.ListenAndServe(":2112", nil)
}
```

It gives me the following metrics:
```
# HELP one_two_three_myapp_processed_ops_total The total number of processed events
# TYPE one_two_three_myapp_processed_ops_total counter
one_two_three_myapp_processed_ops_total 2
# HELP one_two_three_promhttp_metric_handler_errors_total Total number of internal errors encountered by the promhttp metric handler.
# TYPE one_two_three_promhttp_metric_handler_errors_total counter
one_two_three_promhttp_metric_handler_errors_total{cause="encoding"} 0
one_two_three_promhttp_metric_handler_errors_total{cause="gathering"} 0
```

Please let me know if I might still misunderstand something. :)